.github/workflows/system-test.yml

```diff
@@ -0,0 +1,66 @@
+name: system-test
+
+on:
+  pull_request_target:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  is-pr-approved:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: read
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/is-pr-approved@current
+  run-system-tests:
+    needs:
+      - is-pr-approved
+    timeout-minutes: 60
+    permissions:
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.4.6
+      - uses: hashicorp/vault-action@v2.5.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          method: approle
+          secrets: |
+            secret/observability-team/ci/service-account/apm-queue-system-tests credentials | GOOGLE_CREDS ;
```

Comment:
This will not work in forks, it should be in another workflow triggered by `on_workflow`

Reply:
> For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
There is a Organization policy that do es not allow forks to access to secrets, so it will not work.

Reply:
I had to verify this.

https://github.com/elastic/test-pull-request-target/actions/runs/5133114081/jobs/9235276648?pr=2

The secret is hidden with `***` but it seems to work.